src/lib.rs

```diff
@@ -0,0 +1,339 @@
+mod network;
+
+#[allow(dead_code, unused_imports)]
+
+pub type Meters = f32;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum End {
+    Front,
+    Back,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Direction {
```

Comment:
Direction vs Directions vs LaneDirections is going to be confusing in the long-run ;)